src/ucp/api/ucp.h

```diff
@@ -2818,14 +2865,18 @@ ucs_status_ptr_t ucp_am_send_nbx(ucp_ep_h ep, unsigned id,
  * message is delivered to the @a buffer. If the receive operation cannot be
  * started the routine returns an error.
  *
+ * @note This routine can be performed on any valid data descriptor delivered in
+ *       @ref ucp_am_recv_callback_t. Valid data descriptor means that it is either
+ *       rendezvous request (@a UCP_AM_RECV_ATTR_FLAG_RNDV is set in
+ *       @ref ucp_am_recv_param_t.recv_attr) or persistent data pointer
+ *       (@a UCP_AM_RECV_ATTR_FLAG_DATA is set in
+ *       @ref ucp_am_recv_param_t.recv_attr). In the latter case receive operation
+ *       may be needed to unpack data to GPU memory or some specific datatype.
```

Comment:
GPU -> Device memory ? Do we have to explicitly call out GPU by the name ? I would prefer `device memory (for example GPU, DPU, etc)`

Reply:
@tonycurtis I think the definition of `valid` descriptor has to be more specific. I would suggest to create bullet list of properties that make it `valid`. What type of error is reported when `invalid` descriptor is passed.